kernel/aster-nix/src/syscall/select.rs

```diff
@@ -63,7 +63,7 @@ pub fn sys_select(
         Err(e) if e.error() == Errno::ETIME => {
             // The return value is zero if the timeout expires
             // before any file descriptors became ready
-            return Ok(SyscallReturn::Return(0));
+            0
```

Comment:
If ETIME occurs, `readfds`, `writefds` and `exceptfds` should all be empty, right? Why should we write these empty sets back?

Reply:
Yes, all the fds should be empty. However, the user may not check the return value first; instead, they may use `FD_ISSET` to check the fds directly. Here is an example: [lib_timing](https://github.com/asterinas/lmbench/blob/master/src/lib_timing.c#L416)

Also, the [manual](https://www.man7.org/linux/man-pages/man2/select.2.html)  explains the values of the `readfds`, `writefds`, and `exceptfds` after the `select()` function returns. Let's take the `readfds` as an example: "After `select()` has returned, `readfds` will be cleared of file descriptors except for those that are ready for reading."

Reply:
Thanks for your explanation! The change makes sense to me!